docs/sphinx/examples/cpp/providers/nvqc_qml.cpp

```diff
@@ -0,0 +1,90 @@
+// Compile and run with:
+// ```
+// nvq++ --target nvqc nvqc_qml.cpp -o out.x
+// ./out.x
+// ```
+// Assumes a valid NVQC API key has been set in the `NVQC_API_KEY` environment
+// variable. Please refer to the documentations for information about how to
+// attain NVQC API key.
+#include <ctime>
+#include <cudaq.h>
+#include <cudaq/algorithm.h>
+#include <cudaq/optimizers.h>
+#include <random>
+
+struct kernel {
+
+  auto operator()(const int n_qubits,
+                  const std::vector<double> &parameters) __qpu__ {
```

Comment:
Passing by reference isn't really allowed
```suggestion
                  const std::vector<double> parameters) __qpu__ {
```